src/traits.rs

```diff
@@ -1,6 +1,32 @@
 use crate::{IsoWeek, Weekday};
 
 /// The common set of methods for date component.
+///
+/// Methods such as [`year`], [`month`], [`day`] and [`weekday`] can be used to get basic
+/// information about the date.
+///
+/// The `with-*` methods can change the date.
+///
+/// # Warning
+///
+/// The `with-*` methods can be convenient to change a single component of a date. But use them
```

Comment:
Sentences starting with "But" don't read that well... Maybe something like this?

"The `with-*` methods provide a convenient way to change a single component of a date, but they must be used with care. Examples to watch out for:"

I find bullet lists easier to review in code with a blank line before the first bullet (between the "Examples to watch out for:" and the first bullet).